From 8c90042dee7cd0d7960c86b9fcfac5f1afec5784 Mon Sep 17 00:00:00 2001
From: Andrew Gregory <andrew.gregory.8@gmail.com>
Date: Tue, 12 Feb 2013 17:09:49 -0500
Subject: [PATCH] make BINDIR and SBINDIR available to input files

BINDIR and SBINDIR can be set independently of PREFIX and may not be set
to PREFIX/bin as scripts currently assume.

Note from William Hubbs:

This adds @BINDIR@ and @SBINDIR@ macros to the sed commands run to
convert *.in files to the executable form.
Signed-off-by: Andrew Gregory <andrew.gregory.8@gmail.com>
---
 mk/scripts.mk      | 2 +-
 src/librc/Makefile | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/mk/scripts.mk b/mk/scripts.mk
index 7a79821..2201199 100644
--- a/mk/scripts.mk
+++ b/mk/scripts.mk
@@ -12,7 +12,7 @@ _PKG_SED:=		$(shell ${_PKG_SED_SH})
 _LCL_SED_SH=		if test "${PREFIX}" = "${LOCAL_PREFIX}"; then echo "-e 's:@LOCAL_PREFIX@::g'"; else echo "-e 's:@LOCAL_PREFIX@:${LOCAL_PREFIX}:g'"; fi
 _LCL_SED:=		$(shell ${_LCL_SED_SH})
 
-SED_REPLACE=		-e 's:@SHELL@:${SH}:g' -e 's:@LIB@:${LIBNAME}:g' -e 's:@SYSCONFDIR@:${SYSCONFDIR}:g' -e 's:@LIBEXECDIR@:${LIBEXECDIR}:g' -e 's:@PREFIX@:${PREFIX}:g' ${_PKG_SED} ${_LCL_SED}
+SED_REPLACE=		-e 's:@SHELL@:${SH}:g' -e 's:@LIB@:${LIBNAME}:g' -e 's:@SYSCONFDIR@:${SYSCONFDIR}:g' -e 's:@LIBEXECDIR@:${LIBEXECDIR}:g' -e 's:@PREFIX@:${PREFIX}:g' -e 's:@BINDIR@:${BINDIR}:g' -e 's:@SBINDIR@:${SBINDIR}:g' ${_PKG_SED} ${_LCL_SED}
 
 # Tweak our shell scripts
 %.sh: %.sh.in
diff --git a/src/librc/Makefile b/src/librc/Makefile
index 0b4beb2..7307560 100644
--- a/src/librc/Makefile
+++ b/src/librc/Makefile
@@ -18,6 +18,8 @@ SED_CMD=	-e 's:@PREFIX@:${PREFIX}:g'
 SED_CMD+=	-e 's:@LIB@:${LIBNAME}:g'
 SED_CMD+=	-e 's:@SYSCONFDIR@:${SYSCONFDIR}:g'
 SED_CMD+=	-e 's:@LIBEXECDIR@:${LIBEXECDIR}:g'
+SED_CMD+=	-e 's:@BINDIR@:${BINDIR}:g'
+SED_CMD+=	-e 's:@SBINDIR@:${SBINDIR}:g'
 
 _PKG_PREFIX=	-e 's:.*@PKG_PREFIX@.*:\#undef RC_PKG_PREFIX:g'
 ifneq (${PKG_PREFIX},)
-- 
1.8.3

